ttnn/cpp/ttnn/operations/conv/conv2d/device/kernels/reader_conv_activations_2d_mcast_padded_with_halo_3x3_weights_v2.cpp

```diff
@@ -115,22 +139,43 @@ void kernel_main() {
     // Fully create act matrix and tilize it before mcast
     // set_state uses just x/y from the get_noc_addr, addr is ignored
     uint32_t act_l1_read_addr = get_read_ptr(cb_id_sharded_act);
-    noc_async_read_one_packet_set_state(get_noc_addr(act_l1_read_addr), coalesced_read_bytes);
+    if(dilation_w==1)
+    {
+        noc_async_read_one_packet_set_state(get_noc_addr(act_l1_read_addr), coalesced_read_bytes);
+    }
+    else
+    {
+        noc_async_read_one_packet_set_state(get_noc_addr(act_l1_read_addr), conv_act_c_read_bytes);
+    }
 
     // Reset reader_idx to finish act_block_h_datums
     uint32_t reader_idx = 0;
     for (uint32_t nbh = 0; nbh < act_num_blocks_h; nbh++) {
         cb_reserve_back(cb_id_act_row_major_bfloat16, act_block_num_tiles);
         uint32_t l1_write_addr_act = get_write_ptr(cb_id_act_row_major_bfloat16);
 
-        constexpr uint32_t stride_h_bytes = (conv_act_size_w + 2) * conv_act_c_read_bytes;
-        // #pragma GCC unroll 4 // didn't seem to help (neutral), manual unroll 2x perf drop
-        for (uint32_t bh = 0; bh < act_block_h_datums / 2; bh++) {
-            uint32_t two_reader_indices = packed_reader_indices_ptr[reader_idx];
-            read_channels(l1_write_addr_act, act_l1_read_addr, two_reader_indices & 0xffff, conv_act_c_read_bytes, coalesced_read_bytes, stride_h_bytes);
-            read_channels(l1_write_addr_act, act_l1_read_addr, two_reader_indices >> 16   , conv_act_c_read_bytes, coalesced_read_bytes, stride_h_bytes);
+        constexpr uint32_t stride_h_bytes = padded_conv_act_size_w * conv_act_c_read_bytes * dilation_h;
+        constexpr uint32_t stride_w_bytes = conv_act_c_read_bytes * dilation_w;
 
-            reader_idx++;
+        // #pragma GCC unroll 4 // didn't seem to help (neutral), manual unroll 2x perf drop
+        if(dilation_w==1)
```

Comment:
Would it be possible to combine these code paths?

Would performance be an issue?

Reply:
When dilation_w == 1, an entire row of the sliding window can be read contiguously in one read.

When dilation_w > 1, only a single channel can be read contiguously. Therefore `window_width` number of reads are needed for one row.  That's why you will see an additional inner loop for the reader when dilation_w > 1.

Reply:
I would suggest having a single read_channels method then and using `#ifdef` inside of it two have the different paths.
That would require moving dilation info (at least w) to be usable by `#ifdef` which is a slightly bigger change.

@mywoodstock what do you think?